datasetinsights/io/config_handler.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import tempfile
+from pathlib import Path
+
+from yacs.config import CfgNode
+
+import datasetinsights.constants as const
+from datasetinsights.io import create_downloader
+from datasetinsights.io.exceptions import ConfigLoadError, InvalidOverrideKey
+
+logger = logging.getLogger(__name__)
+_EQUAL_STR = "="
+
+
+def prepare_config(path=None, override=None):
+    """load and override config from local or remote locations .
+
+    Args:
+        path: config location
+        override: key-value pairs to override config
+    Returns:
+        config object of type yacs.config.CfgNode
+    """
+    config = load_config(path)
+    logger.info(f"config loading from {path} completed")
+    if override:
+        logger.info(f"overriding config params= {override}")
+        override_config(override=override, config=config)
+    return config
+
+
+def load_config(path):
+    """ load config from local or remote locations .
+    Examples:
+        >>> path="gs://thea-dev/../config.yaml"
+        >>> path="https://thea-dev/../config.yaml"
+        >>> path="http://thea-dev/../config.yaml"
+        >>> path="file:///root/../config.yaml" # absolute path
+        >>> path="file://datasetinsights/../config.yaml" # relative local path
+
+    Returns:
+        config object of type yacs.config.CfgNode
+    """
+    if not path.startswith(const.LOCAL_FILE_BASE_STR):
+        try:
+            downloader = create_downloader(source_uri=path)
+            with tempfile.TemporaryDirectory() as tmp:
+                downloader.download(source_uri=path, output=tmp)
+                logger.debug(f"downloading to directory: {tmp}")
+                config_path = Path(path)
+                file_path = tmp / config_path.relative_to(config_path.parent)
+                return CfgNode.load_cfg(open(file_path, "r"))
+        except ValueError:
+            logger.exception(f"Prefix does not support for path: {path}")
+            raise ValueError
+        except Exception:
```

Comment:
You should catch this exception error message and send it to logs here. It should be helpful to let user know what's the root cause of exceptions rather than hiding it. For example:
```
except Exception as e: 
    logger.execption("some message")
    logger.exception(e)
    raise  
```